tests/callbacks/test_callbacks.py

```diff
@@ -11,168 +11,12 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-from unittest import mock
-from unittest.mock import ANY, call, MagicMock, Mock
+from unittest.mock import call, Mock
 
 from pytorch_lightning import Trainer
 from tests.helpers import BoringModel
 
 
-@mock.patch("torch.save")  # need to mock torch.save or we get pickle error
-def test_trainer_callback_hook_system_fit(_, tmpdir):
```

Comment:
why do you delete this? this checks the exact call order plus the called args.

Reply:
Because the order is already tested with the new changes. See rest of the PR.

Not the arguments though, but I don't think this is useful considering we were just checking the trainer and model, everything else is set to ANY

Reply:
The number of arguments change frequently because we add and remove and deprecate all the time some outputs or inputs. These tests will fail and alert breaking changes, but the others in HookedModel not. 

The tests that rely on helper callback to append a string in a list rely on the correctness of that implementation. This is not ideal, as you would need a test for that to be correct as well. I hope you understand what I mean.

Reply:
How would you test the order of the LightningModule and Callback together?

Reply:
By attaching the two mocks (one for callback and one for module) to a third one that tracks all calls in order. 
Something like this:
https://stackoverflow.com/questions/22677280/checking-call-order-across-multiple-mocks

I'm not sure if it works with our objects but I would like to try it if you don't delete the test we can eventually extend it. I strongly believe we should not rely on our own internal testing frameworks. The same can be said about the dev debugger.

Reply:
> By attaching the two mocks (one for callback and one for module) to a third one that tracks all calls in order.

To do so, we would need to be able to mock the `LightningModule`. Tried unsuccessfully here: 

https://github.com/PyTorchLightning/pytorch-lightning/compare/tests/lightningmodule-mock

---

I had the idea to use `LambdaCallback` for the hooked model, this way the addition of any hooks will break the test and we can include the arguments for each hook and test those too